### ISSUE TITLE
Automatically set GOMAXPROCS and GOMEMLIMIT

### ISSUE THREAD

**Issue by jenshu** [solo-io/gloo#10062]
### Gloo Edge Product

Open Source

### Gloo Edge Version

1.17

### Is your feature request related to a problem? Please describe.

It is good practice to set the Go runtime environment variables GOMAXPROCS and GOMEMLIMIT, which tell Go the max threads/memory it should use. We should set these in our containers that run Go (namely gloo and discovery) whenever resource limits are set.

See:
https://pkg.go.dev/runtime
https://blog.howardjohn.info/posts/gomaxprocs/

┆Issue is synchronized with this [Asana task](https://app.asana.com/0/1206768562311555/1208338214628331) by [Unito](https://www.unito.io)

**Comment by jenshu** [comment]
per internal discussion, we will merge to 1.18 only (no backport)

**Comment by jenshu** [comment]
If this is needed in GlooEE 1.17 and earlier, the env variables can be added via helm (only set these variables if you are setting resource limits in the gloo pod):
```
gloo:
  gloo:
    deployment:
      customEnv:
      - name: GOMEMLIMIT
        valueFrom:
          resourceFieldRef:
            divisor: "1"
            resource: limits.memory
      - name: GOMAXPROCS
        valueFrom:
          resourceFieldRef:
            divisor: "1"
            resource: limits.cpu
```

**Comment by jenshu** [comment]
will be available in OSS 1.18.0-beta23 and EE 1.18.0-beta2
